applications/ExaquteSandboxApplication/custom_processes/divergence_process.h

```diff
@@ -0,0 +1,217 @@
+#include "processes/process.h"
+#include "utilities/variable_utils.h"
+#include "exaqute_sandbox_application_variables.h"
+
+// Application includes
+
+namespace Kratos
+{
+    ///@addtogroup ExaquteSandboxApplication
+    ///@{
+
+    ///@name Kratos Globals
+    ///@{
+
+    ///@}
+    ///@name Type Definitions
+    ///@{
+
+    ///@}
+    ///@name  Enum's
+    ///@{
+
+    ///@}
+    ///@name  Functions
+    ///@{
+
+    ///@}
+    ///@name Kratos Classes
+    ///@{
+
+    /// Process to compute divergence
+    /**
+     * This process computes the divergence and the seminorm of the velocity field.
+     * We define VELOCITY_H1_SEMINORM as: \left \| \nabla u_{h} \right \|_{L^2(K)}^2 ,
+     * and DIVERGENCE as \left \| \nabla \cdot u_{h} \right \|_{L^2(K)}^2 ,
+     * where u is the velocity field and K an element of the domain \Omega.
+     */
+    class KRATOS_API(EXAQUTE_SANDBOX_APPLICATION) DivergenceProcess : public Process
+    {
+    public:
+        ///@name Type Definitions
+        ///@{
+
+        /// Pointer definition of Process
+        KRATOS_CLASS_POINTER_DEFINITION(DivergenceProcess);
+
+        ///@}
+        ///@name Life Cycle
+        ///@{
+
+        /// Default constructor
+        /**
+         * @brief Construct DivergenceProcess object
+         * @param rModelPart Model part the process is applied to
+         */
+        DivergenceProcess(
+        ModelPart& rModelPart);
+
+        /// Destructor.
+        ~DivergenceProcess() override = default;
+
+        /// Assignment operator.
+        DivergenceProcess& operator=(DivergenceProcess const& rOther) = delete;
+
+        ///@}
+        ///@name Operators
+        ///@{
+
+        ///@}
+        ///@name Operations
+        ///@{
+
+        /**
+         * @brief Function initializing the process
+         */
+        void ExecuteInitialize() override;
+
+        /**
+         * @brief Function computing quantities at each time step
+         */
+        void ExecuteBeforeOutputStep() override;
+
+        ///@}
+        ///@name Access
+        ///@{
+
+        ///@}
+        ///@name Inquiry
+        ///@{
+
+        ///@}
+        ///@name Input and output
+        ///@{
+
+        /// Turn back information as a string.
+        std::string Info() const override;
+
+        /// Print information about this object.
+        void PrintInfo(std::ostream& rOStream) const override;
+
+        /// Print object's data.
+        void PrintData(std::ostream& rOStream) const override;
+
+        ///@}
+        ///@name Friends
+        ///@{
+
+        ///@}
+
+        protected:
+        ///@name Protected static Member Variables
+        ///@{
+
+
+        ///@}
+        ///@name Protected member Variables
+        ///@{
+
+
+        ///@}
+        ///@name Protected Operators
+        ///@{
+
+
+        ///@}
+        ///@name Protected Operations
+        ///@{
+
+        double ComputeAuxiliaryElementDivergence(Vector& grad_x, Vector& grad_y, Vector& grad_z);
```

Comment:
Doxygen documentation is missing.

Reply:
I also think that these two methods can be `private`.

Reply:
Added and moved to private.